src/render_cmd.rs

```diff
@@ -171,12 +180,28 @@ impl RenderCmd {
             }
             cmd[2] = release_name.to_owned();
 
-            let fully_qualified_output_dir = format!("{}/{}/{}", output_dir, d.name, release_name);
-            cmd.push(format!("--output-dir={}", fully_qualified_output_dir));
 
-            plan.commands.insert(d.name.to_owned(), cmd);
-            plan.output_paths
-                .insert(d.name.clone(), PathBuf::from(fully_qualified_output_dir));
+            // if the filter flag is passed, include only the filtered deployment. Otherwise allow all
+            match &self.opts.filter {
```

Comment:
I reckon you already noticed that this duplicates the existing code. Since we are in a loop at this point, I propose to try something like:

```rust
if &self.opts.filter.is_some() && !self.is_name_filetered() {
  continue;
}
```

This way you can avoid to duplicate the code and increase readability.

Reply:
The loop is not supposed to exist if the condition is not met. In either case (both when we have filter option and when we don't), we want to populate `Plan`. The difference is the value of `name` we are passing.

Reply:
> The loop is not supposed to exist if the condition is not met.

I do not understand what you mean. The loop is always there as we iterate over all deployments to populate the `Plan`.

Reply:
We're in the scope if this loop:

```rust
for d in &cfg.deployments {
  // pseudo code
  IF we have a --filter AND current deployment does not match THEN
    continue
  ENDIF
}
```